modules/light-clients/07-tendermint/light_client_module.go

```diff
@@ -295,30 +303,45 @@ func (lcm LightClientModule) VerifyUpgradeAndUpdateState(
 		return errorsmod.Wrapf(clienttypes.ErrInvalidClientType, "expected: %s, got: %s", exported.Tendermint, clientType)
 	}
 
-	var newClientState ClientState
-	if err := lcm.keeper.Codec().Unmarshal(newClient, &newClientState); err != nil {
+	var (
+		cdc               = lcm.keeper.Codec()
+		newClientState    exported.ClientState
+		newConsensusState exported.ConsensusState
+	)
+
+	if err := cdc.UnmarshalInterface(newClient, &newClientState); err != nil {
 		return err
 	}
-
+	newTmClientState, ok := newClientState.(*ClientState)
+	if !ok {
+		return errorsmod.Wrapf(clienttypes.ErrInvalidClient, "expected client state type %T, got %T", (*ClientState)(nil), newClientState)
+	}
 	if err := newClientState.Validate(); err != nil {
```

Comment:
@colin-axner @damiannolan This validation fails if the trust level is not within [0, 1/3] (see test failing [here](https://github.com/cosmos/ibc-go/actions/runs/7849754768/job/21423772705?pr=5827)). However, the upgraded client state that the counterparty chain commits should have all the custom fields zeroed out [since that's what we verify in `VerifyUpgradeAndUpdateState` function in the `ClientState`](https://github.com/cosmos/ibc-go/blob/d957ef80d72a260af3f6e66bbf5d7bea85473e2b/modules/light-clients/07-tendermint/upgrade.go#L73). So I think we either cannot do this validation here or we need to validate only the fields that shouldn't be zeroed out.

Reply:
I saw this failing test when working on the one above. I didn't dig any deeper on this one yet. But from what you're saying here it sounds correct. I'm not sure exactly what pieces of code we'll need to change yet, can follow up o this one next week! :)

Reply:
I removed the basic validation of client state and consensus state at the light client module level. ClientState is reconstructed within the method `VerifyUpgradeAndUpdateState` and validated there. 

There is still some test failures wrt encoding from what I can see, but I think they should be manageable. Is the plan to restructure tests to invoke the client module in this PR or to leave it for a follow up? 

Happy to help out where needed :)